kstests-in-cloud.sh

```diff
@@ -214,22 +209,34 @@ INVENTORY=${INVENTORY_DIR}/${TARGET}.inventory
 TARGET_KEY_DIR=${STORED_PRIVATE_KEYS_DIR}/${TARGET}
 
 
-# We were scheduled, just modify the "schedule" from scheduling cmdline to "test"
-# and test!
-if [[ $SCHEDULED == "yes" ]]; then
-    COMMAND="test"
-fi
-
 #################################################### schedule test
 
 if [[ ${COMMAND} == "schedule" ]]; then
 
     if [[ ${REMOVE_SCHEDULE} == "yes" ]]; then
         ansible-playbook linchpin/remove_scheduled_tests.yml --extra-vars "test_id=${TARGET}"
     else
-        CMDLINE="\"$0 ${ARGUMENTS} --scheduled\""
 
-        echo ${CMDLINE}
+        if [[ -n ${TEST_CONFIGURATION_FILE} ]]; then
+            TEST_CONFIGURATION_FILE_ARG="--test-configuration ${TEST_CONFIGURATION_FILE}"
+        fi
+        if [[ -n ${RESULTS_DIR} ]]; then
+            RESULTS_DIR_ARG="--results ${RESULTS_DIR}"
+        fi
+        if [[ ${USE_KEY_FOR_MASTER} == "yes" ]]; then
+            USE_KEY_FOR_MASTER_ARG="--key-use-for-master"
+        fi
+        if [[ ${KEY_MODE} == "upload" ]]; then
+            KEY_MODE_ARG="--key-upload"
+        elif [[ ${KEY_MODE} == "existing" ]]; then
+            KEY_MODE_ARG="--key-use-existing"
+        fi
+        if [[ -n ${PRIVATE_KEY_PATH} ]]; then
+            PRIVATE_KEY_PATH_ARG="--ansible-private-key ${PRIVATE_KEY_PATH}"
+        fi
+
+
+        CMDLINE="\"$0 test ${TARGET} --cloud ${CLOUD_PROFILE} --pinfile ${PINFILE} --remote-user ${REMOTE_USER} --key-name ${KEY_NAME} ${KEY_MODE_ARG} ${PRIVATE_KEY_PATH_ARG} ${TEST_CONFIGURATION_FILE_ARG} ${RESULTS_DIR_ARG} ${USE_KEY_FOR_MASTER_ARG}\""
```

Comment:
I'm just curios but do I understand it correctly that you are doing here recursion call?

Reply:
The CMDLINE - yes, the linchpin script call - (basically the original command with `schedule` replaced by `test`) will by called from generated (by the schedule_test playbook) script that would be run by systemd timer/service.

Originally I wanted to be cleverly lazy and just take the cmdline as is and replace `schedule` with `test` in the call from generated script (or - because replacing an argument in cmdline correctly is not trivial - just append private --scheduled option to tell the linchpin script that now it should not schedule but actually run the test). But finally it was too difficult to escape --when argument containing spaces when passing it to the generated script via playbook (using --extra-vars) so I am just creating the CMDLINE from the scratch. Being explicit here requires more bash lines (and may need get updated if additional test options are added) but I think it is acceptable.

Reply:
This will work and I don't want to hold this from merging. However, I would be cautious about recursion calls in bash, it will be hell to debug an issue here.

Reply:
Oh I see, thanks for explanation!